bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/core/IBigtableDataClient.java

```diff
@@ -41,17 +41,17 @@
    * Mutate a row atomically.
    *
    * @param rowMutation a {@link RowMutation} model object.
-   * @return a {@link ApiFuture} of type {@link Void} will be set when request is
+   * @return a {@link ListenableFuture} of type {@link Void} will be set when request is
    *     successful otherwise exception will be thrown.
    * @throws InterruptedException if any.
    */
-  ApiFuture<Void> mutateRowAsync(RowMutation rowMutation) throws InterruptedException;
+  ListenableFuture<Void> mutateRowAsync(RowMutation rowMutation) throws InterruptedException;
```

Comment:
why you converted to ListenableFuture? all GAX uses ApiFuture I think.

Reply:
Because once we moved to GAX, then we can convert its ApiFuuture response to  ListenableFuture... with all GAX models using ApiFutureToListenableFuture.

Also we Cloud-bigtable-client has all it's logic built upon ListenableFuture... if we can continue with existing logic if we go with this.
else 
we will have to change various logic as well (that can also be done but we would require to adapt to ApiFuture in groups as some methods are interdependent like readModifyRow & readFlatRows & bulkMutation in BatchExecutor.java).